web/packages/teleport/src/Account/ManageDevices/AddAuthDeviceWizard/AddAuthDeviceWizard.tsx

```diff
@@ -0,0 +1,332 @@
+        // Step properties
+        privilegeToken={privilegeToken}
+        credential={credential}
+        onClose={onClose}
+        onSuccess={onSuccess}
+        onAuthenticated={onAuthenticated}
+        onPasskeyCreated={onPasskeyCreated}
+      />
+    </Dialog>
+  );
+}
+
+const wizardFlows = {
+  withReauthentication: [
+    ReauthenticateStep,
+    CreatePasskeyStep,
+    SavePasskeyStep,
+  ],
+  withoutReauthentication: [CreatePasskeyStep, SavePasskeyStep],
+};
+
+interface ReauthenticateStepProps extends StepComponentProps {
+  onAuthenticated(privilegeToken: string): void;
+  onClose(): void;
+}
+
+function ReauthenticateStep({
+  next,
+  refCallback,
+  stepIndex,
+  flowLength,
+  onClose,
+  onAuthenticated: onAuthenticatedProp,
+}: ReauthenticateStepProps) {
+  const challengeScope = MfaChallengeScope.MANAGE_DEVICES;
+  const onAuthenticated = (privilegeToken: string) => {
+    onAuthenticatedProp(privilegeToken);
+    next();
+  };
+  const {
+    attempt,
+    auth2faType,
+    preferredMfaType,
+    clearAttempt,
+    submitWithTotp,
+    submitWithWebauthn,
+  } = useReAuthenticate({
+    onAuthenticated,
+    challengeScope,
+  });
+  const mfaOptions = createMfaOptions({
+    auth2faType,
+    preferredType: preferredMfaType,
+    required: true,
+  });
+
+  const [mfaOption, setMfaOption] = useState<Auth2faType>(mfaOptions[0].value);
+  const [authCode, setAuthCode] = useState('');
+
+  const onAuthCodeChanged = (e: React.ChangeEvent<HTMLInputElement>) => {
+    setAuthCode(e.target.value);
+  };
+
+  const onReauthenticate = (
+    e: FormEvent<HTMLFormElement>,
+    validator: Validator
+  ) => {
+    e.preventDefault();
+    if (!validator.validate()) return;
+    if (mfaOption === 'webauthn') {
+      submitWithWebauthn(challengeScope);
+    }
+    if (mfaOption === 'otp') {
+      submitWithTotp(authCode);
+    }
+  };
+
+  return (
+    <div ref={refCallback} data-testid="reauthenticate-step">
+      <Text typography="body2">
+        Step {stepIndex + 1} of {flowLength}
+      </Text>
+      <Text typography="h4">Verify Identity</Text>
+      {attempt.status === 'failed' && (
+        <OutlineDanger>{attempt.statusText}</OutlineDanger>
+      )}
+      Multi-factor type
+      <Validation>
+        {({ validator }) => (
+          <form onSubmit={e => onReauthenticate(e, validator)}>
+            <RadioGroup
+              name="mfaOption"
+              options={mfaOptions}
+              value={mfaOption}
+              flexDirection="row"
+              gap={3}
+              onChange={o => {
+                setMfaOption(o as Auth2faType);
+                clearAttempt();
+              }}
+            />
+            {mfaOption === 'otp' && (
+              <FieldInput
+                label="Authenticator Code"
+                rule={requiredField('Authenticator code is required')}
+                inputMode="numeric"
+                autoComplete="one-time-code"
+                value={authCode}
```

Comment:
I'm wondering if we should introduce a new name, I believe we use 'otp code' everywhere else.

Reply:
No, there's actually a discrepancy between the Web UI and the CLI. The change is consistent with what were doing until now.